akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardCoordinatorDowning2Spec.scala

```diff
@@ -0,0 +1,181 @@
+
+class PersistentClusterShardCoordinatorDowning2MultiJvmNode1 extends PersistentClusterShardCoordinatorDowning2Spec
+class PersistentClusterShardCoordinatorDowning2MultiJvmNode2 extends PersistentClusterShardCoordinatorDowning2Spec
+
+class DDataClusterShardCoordinatorDowning2MultiJvmNode1 extends DDataClusterShardCoordinatorDowning2Spec
+class DDataClusterShardCoordinatorDowning2MultiJvmNode2 extends DDataClusterShardCoordinatorDowning2Spec
+
+abstract class ClusterShardCoordinatorDowning2Spec(multiNodeConfig: ClusterShardCoordinatorDowning2SpecConfig)
+    extends MultiNodeClusterShardingSpec(multiNodeConfig)
+    with ImplicitSender {
+  import multiNodeConfig._
+
+  import ClusterShardCoordinatorDowning2Spec._
+
+  def startSharding(): Unit = {
+    startSharding(
+      system,
+      typeName = "Entity",
+      entityProps = Props[Entity](),
+      extractEntityId = extractEntityId,
+      extractShardId = extractShardId)
+  }
+
+  lazy val region = ClusterSharding(system).shardRegion("Entity")
+
+  s"Cluster sharding ($mode) with down member, scenario 2" must {
+
+    "join cluster" in within(20.seconds) {
+      startPersistenceIfNotDdataMode(startOn = first, setStoreOn = Seq(first, second))
+
+      join(first, first, onJoinedRunOnFrom = startSharding())
+      join(second, first, onJoinedRunOnFrom = startSharding(), assertNodeUp = false)
+
+      // all Up, everywhere before continuing
+      runOn(first, second) {
+        awaitAssert {
+          cluster.state.members.size should ===(2)
+          cluster.state.members.unsorted.map(_.status) should ===(Set(MemberStatus.Up))
+        }
+      }
+
+      enterBarrier("after-2")
+    }
+
+    "initialize shards" in {
+      runOn(first) {
+        val shardLocations = system.actorOf(Props[ShardLocations](), "shardLocations")
+        val locations = (for (n <- 1 to 4) yield {
+          val id = n.toString
+          region ! Ping(id)
+          id -> expectMsgType[ActorRef]
+        }).toMap
+        shardLocations ! Locations(locations)
+        system.log.debug("Original locations: {}", locations)
+      }
+      enterBarrier("after-3")
+    }
+
+    "recover after downing other node (not coordinator)" in within(20.seconds) {
+      val secondAddress = address(second)
+
+      runOn(first) {
+        testConductor.blackhole(first, second, Direction.Both).await
+      }
+
+      Thread.sleep(3000)
+
+      runOn(first) {
+        cluster.down(second)
+        awaitAssert {
+          cluster.state.members.size should ===(1)
+        }
+
+        // start a few more new shards, could be allocated to second but should notice that it's terminated
+        val additionalLocations =
+          awaitAssert {
+            val probe = TestProbe()
+            (for (n <- 5 to 8) yield {
+              val id = n.toString
+              region.tell(Ping(id), probe.ref)
+              id -> probe.expectMsgType[ActorRef](1.second)
```

Comment:
With this test I could reproduce the issue. It still works with the watch fix in ClusterRemoteWatcher without changes to the ShardCoordinator, but without that it fails. With the additional improvement to the ShardCoordinator it also works without the watch fix.

Reply:
Better to avoid doing it rather than relying on the watch 👍